corehq/apps/accounting/forms.py

```diff
@@ -1743,15 +1746,39 @@ def amount(self):
                                   % adjustment_type)
 
     def adjust_balance(self, web_user=None):
-        CreditLine.add_credit(
-            -self.amount,
-            account=self.invoice.subscription.account,
-            subscription=self.invoice.subscription,
-            note=self.cleaned_data['note'],
-            invoice=self.invoice,
-            reason=self.cleaned_data['method'],
-            web_user=web_user,
-        )
+        method = self.cleaned_data['method']
+        kwargs = {
+            'account': self.invoice.subscription.account,
+            'note': self.cleaned_data['note'],
+            'reason': method,
+            'subscription': self.invoice.subscription,
+            'web_user': web_user,
+        }
+        if method == CreditAdjustmentReason.MANUAL:
+            CreditLine.add_credit(
+                -self.amount,
```

Comment:
The balance is calculated by summing the record of credit adjustments, so there's a need to create 2 records to avoid a side effect.

It is true that we could wrap these 2 function calls into a single ```CreditLine.add_to_invoice``` function